metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -437,7 +551,24 @@ public void rename(
         if (oldTable != null) {
             //Ignore if the operation is not supported, so that we can at least go ahead and save the user metadata
             eventBus.post(new MetacatRenameTablePreEvent(oldName, metacatRequestContext, this, newName));
-            connectorTableServiceProxy.rename(oldName, newName, isMView);
+
+            // Before rename, first rename its parent child relation
```

Comment:
making a note this will change if we make that adjustment to duplicate at first step of rename as discussed offline

Reply:
we can also utilize the callback structure you have above for rolling back the create

Reply:
For the rename, I will put up with a separate PR, thanks!